core/arch/arm/kernel/generic_entry_a64.S

```diff
@@ -137,11 +138,13 @@ clear_nex_bss:
 	/* Enable Console */
 	bl	console_init
 
+	adr	x0, boot_mmu_config
 	bl	core_init_mmu_map
-	bl	core_init_mmu_regs
-	bl	cpu_mmu_enable
-	bl	cpu_mmu_enable_icache
-	bl	cpu_mmu_enable_dcache
+
+	bl	__get_core_pos
+	bl	enable_mmu
+	/* Virtual addresses may have changed, reinitialize console */
```

Comment:
No, this is old stuff needed without ASLR.

Reply:
I don't understand why it is needed, with or without ASLR actually. The console code uses a PA when the MMU is disabled, then resolves the PA to a VA (and caches it) when the MMU is enabled (thanks to `io_pa_or_va()`). So, as long as the VA does not change, we should be good. 

What do you mean by "Virtual addresses may have changed"? What could change the mapping once it has been established and the MMU turned on?

I must be missing something because if I remove that `console_init`, all is fine w/o ASLR but the initialization hangs with ASLR :-/

Reply:
The problem is with `register_serial_console()` which has previously registered a physical address, now that virtual addresses are different from physical ones we need to register again.

I'll guard the call with `CFG_CORE_ASLR`.

Reply:
That's it... I wasn't looking at the right place, I was too focused on this `io_pa_or_va()` thing :-/ Would you mind updating the comment to something like this?
```C 
 /*
  * Reinitialize console, since register_serial_console() has previously
  * registered a PA and with ASLR the VA is different from the PA.
  */
```

Reply:
OK